app.py

```diff
@@ -46,7 +46,14 @@ def inject_configurations():
 @app.route('/')
 def index():  # put application's code here
     if session and session['logged_in']:
-        return render_template("home.html")
+        user = db.session.query(User).filter_by(email=session['email']).first()
+        courses = db.session.query(Course).all()
+        teachers_course = {
```

Comment:
Can't you use the backrefs from the Teacher model to the courses (and therefore an SQL request), instead of filtering manually in a Python for loop ?